rclcpp/src/rclcpp/rate.cpp

```diff
@@ -67,7 +67,8 @@ Rate::sleep()
   // Calculate the time to sleep
   auto time_to_sleep = next_interval - now;
   // Sleep (will get interrupted by ctrl-c, may not sleep full time)
-  return clock_->sleep_for(time_to_sleep);
+  clock_->sleep_for(time_to_sleep);
+  return true;
```

Comment:
I still think that this is the way we could take. but irrelevant to this flaky test reason.

since time goes by during calling `sleep_for` method. if that is racy, it actually does not sleep in here.

Reply:
Okay, let's do this, since in original code,the return value of `rclcpp::sleep_for()` [was ignored](https://github.com/ros2/rclcpp/blob/769a9d0439afade4f526acdc9588a8c15c88f330/rclcpp/include/rclcpp/rate.hpp#L86), we'd better to follow this logic in new code, as well